src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -114,22 +116,40 @@ public virtual void SetupChannel(IEnumerable<string> sources)
                     // Increase connection timeout when debugging is enabled.
                     connTimeout = 5 * this.connectionTimeout;
                 }
-                
-                this.initialized = true;
-            }
 
-            // Wait for a timeout for the client to connect.
-            if (!this.RequestSender.WaitForRequestHandlerConnection(connTimeout))
-            {
-                var errorMsg = CrossPlatEngineResources.InitializationFailed;
+                // Wait for a timeout for the client to connect.
+                if (!this.RequestSender.WaitForRequestHandlerConnection(connTimeout))
+                {
+                    var errorMsg = CrossPlatEngineResources.InitializationFailed;
+
+                    if (!string.IsNullOrWhiteSpace(this.testHostProcessStdError.ToString()))
+                    {
+                        // Testhost failed with error
+                        errorMsg = string.Format(CrossPlatEngineResources.TestHostExitedWithError, this.testHostProcessStdError);
+                    }
 
-                if (!string.IsNullOrWhiteSpace(this.testHostProcessStdError))
+                    throw new TestPlatformException(string.Format(CultureInfo.CurrentUICulture, errorMsg));
+                }
+
+                // Handling special case for dotnet core projects with older test hosts
+                // Older test hosts are not aware of protocol version check
+                // Hence we should not be sending VersionCheck message to these test hosts
+                bool checkRequired = true;
+                var property = this.testHostManager.GetType().GetRuntimeProperties().FirstOrDefault(p => string.Equals(p.Name, versionCheckPropertyName, StringComparison.OrdinalIgnoreCase));
+                if (property != null)
```

Comment:
Don't really know if there is something better we can do here. Seems odd :(
Maybe have a config file with the version and check for its existence?